Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -178,6 +179,14 @@ import Foundation
                         graphQLHTTP = BTGraphQLHTTP(url: graphQLBaseURL, tokenizationKey: tokenizationKey)
                     }
                 }
+                
+                if paypalHTTP == nil {
+                    let paypalBaseURL: URL? = paypalAPIURL(forEnvironment: configuration?.environment ?? "")
```

Comment:
Do we want to make this method return a non-optional URL? Based on this method it seems like the default is the prod URL. Based on the defaulting behavior it may also make sense to make the environment optional instead of passing an empty string here if we don't get one from the config. Thoughts?

Reply:
Ah - good ❓. I was following the existing pattern that we have for the GraphQL API implementation ([see here](https://github.com/braintree/braintree_ios/blob/c90a487da1b93fb0790c1fd6cca61267fb90b42b/Sources/BraintreeCore/BTAPIClient.swift#L173)).

I agree though, it's definitely funky.

Wrote this in the PR notes a bit, but this PR revealed how tightly coupled our networking layer is to BT APIs. Even the GraphQL stuff feels very tacked-on. I think we could benefit from a re-design, but wanted to 🏈 on it.